sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -310,15 +312,15 @@ def __init__(
         environment_payload = proto_utils.parse_Bytes(
             environment.payload, beam_runner_api_pb2.DockerPayload)
         container_image_url = environment_payload.container_image
-        if container_image_url == pipeline_sdk_container_image:
-          # This was already added
+        if container_image_url in already_added_containers:
+          # Do not add the pipeline environment again.
```

Comment:
Perhaps also add a comment that currently dataflow stages all dependencies to all environments?

Reply:
Sent https://github.com/apache/beam/pull/11757.